velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -456,5 +456,52 @@ TEST_F(DateTimeFunctionsTest, quarterDate) {
   EXPECT_EQ(3, quarter("1954-08-08"));
 }
 
+TEST_F(DateTimeFunctionsTest, nextDay) {
+  const auto nextDay = [&](const std::string& date,
+                           const std::string& dayOfWeek) {
+    auto startDates =
+        makeNullableFlatVector<int32_t>({parseDate(date)}, DATE());
+    auto dayOfWeeks = makeNullableFlatVector<std::string>({dayOfWeek});
+
+    auto result = evaluateOnce<int32_t>(
+        fmt::format("next_day(c0, '{}')", dayOfWeek),
+        makeRowVector({startDates}));
+
+    auto anotherResult = evaluateOnce<int32_t>(
+        "next_day(c0, c1)", makeRowVector({startDates, dayOfWeeks}));
+
+    EXPECT_EQ(result, anotherResult);
+    std::optional<std::string> res;
+    if (result.has_value()) {
```

Comment:
EXPECT_EQ(result.has_value())

Reply:
@mbasmanova Changing this to ```EXPECT_EQ``` will cause an exception to be thrown directly when the format is invalid, ```nextDay("2015-07-23", "xx")```cannot return null, causing test cases with invalid formats to fail, such as```EXPECT_EQ(nextDay("2015-07-23", "xx"), std::nullopt);```.